### PR TITLE
Ks hk filter search

### LINKED ISSUE #9: 9. As a user, I want to filter my shopping list to make it easier to locate an item in the list.
AC:

* Display a text field above the top of the shopping list
* As the user types into the field, the list should narrow to display only items that contain the text the user entered in the filter field
* When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field
* The filter text should match any part of the item name (i.e. it should _not_ only match from the start of the string)

Notes:

* The wireframe shows the filter field, but not it being used. https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png
* The following example app from MDN is a great example of providing feedback that is accessible to screenreaders when things change on the screen. https://moz-todomvc-react.netlify.app/

### PR DESCRIPTION
**Pull request by ksiman14**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

- Added input field where users can type to filter
- Added "X" button that appears when input field isn't empty so user can clear the field
- Created state for search that updates with text in input field
- Typing into the field filters list items to only show items containing the text in the input field

## Related Issue

closes #9 

## Acceptance Criteria

- [x] Display a text field above the top of the shopping list
- [x] As the user types into the field, the list should narrow to display only items that contain the text the user entered in the filter field
- [x] When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field
- [x] The filter text should match any part of the item name (i.e. it should not only match from the start of the string)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Screen Shot 2021-08-09 at 3 32 06 PM

### After

Screen Shot 2021-08-09 at 3 30 08 PM

## Testing Steps / QA Criteria

- Pull down branch `ks-hk-filterSearch`
- Type in search bar
- Items should filter according to text
- X button should appear, click button to clear input

### CODE REVIEW

**Review by isaabutaa — APPROVED** [review]
Very nicely done @hulyak and @ksiman14! I like how you chained the `filter()` and `map()` methods. And everything about the input and `handleChange` method looks great. 
Also, on line 8 of the `AddItem` component, I like how you adjusted the `lastPurchasedDate` variable declaration... Makes sense. 💯

**Review by sandaiiyahh — APPROVED** [review]
I was amazed when you both showed this during office hours yesterday, and it's cool to try it for myself now as well! 😍 Great job @ksiman14 and @hulyak. This seems fun to have coded together!

Really, really cool execution! Thanks for cleaning up the `setLastPurchasedDate` too 👍

File: src/components/AddItem.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { useHistory } from 'react-router-dom';
 const AddItem = () => {
   const [item, setItem] = useState('');
   const [frequency, setFrequency] = useState(0);
-  const [lastPurchasedDate, setLastPurchasedDate] = useState(null);
+  const [lastPurchasedDate] = useState(null);
```
Comment (luisaugusto) [inline src/components/AddItem.js]: Won't this always just be a `null` value? If so, would it make sense to remove this state completely?

File: src/components/ItemsList.js [hunk 2]
```diff
@@ -33,9 +44,13 @@ const ItemsList = () => {
             </>
           ) : (
             <>
-              {snapshot.docs.map((doc, index) => (
-                <SingleItem key={index} {...doc.data()} id={doc.id} />
-              ))}
+              {snapshot.docs
+                .filter((doc) =>
+                  doc.data().name.toLowerCase().includes(search.toLowerCase()),
+                )
```
Comment (luisaugusto) [inline src/components/ItemsList.js:50]: Nice job with this filter! 🚀

**Review by luisaugusto — APPROVED** [review]
Nice job with this! I played around with it and wasn't able to find any issues with it. I just had one comment, but otherwise it's good to go!

**Review by meganesu — APPROVED** [review]
This looks great! I like how you used the `filter` array method. It's super clean and easy to read! 👏

File: src/components/AddItem.js (hunk 1 above)
Comment (ksiman14) [reply src/components/AddItem.js]: You're right! I'm going to remove it, adjust the item template to just use lastPurchasedDate: null and re-commit before merging.

**Pull request merged**